book/website/reproducible-research/licensing.md

```diff
@@ -10,102 +13,301 @@ No previous knowledge is needed, this chapter explains how important it is to un
+    - [Responsible AI Licenses (RAIL)](https://www.licenses.ai/)
+
+(rr-licensing-enforcement)=
+### Licencing enforcement
+
+There have been a number of successful legal cases that have been brought in defence of the terms of copyleft licenses obliging the parties abusing the terms of these licenses to appropriately release their code.
+But this can be hard to discover, as it is not immediately obvious if copyleft code has been used from looking at a black box proprietary end product.
+
+<!-- citations needed -->
+
+Organisations which take legal action in defence of free software, and which can provide information and resources for anyone else seeking to do the same, include:
 
-In broad terms copyleft licenses impose an obligation of reciprocity on users of any of the licensed items. 
-If something is licensed under a copyleft license, then any derivative works need to be shared under the same rights.
-Note that it can make things difficult if you want to combine different elements having different copy-left licenses.
-Most copyleft licenses fall under open source licenses but not all open source licenses are copyleft.
+- [Software Freedom concervancy](https://sfconservancy.org/)
+- [Software Freedom Law Centre](https://softwarefreedom.org/)
+- [FSF - licensing and compliance](https://www.fsf.org/licensing/)
+- [Free Software Foundation Europe (FSFe) - legal work](https://fsfe.org/activities/legal.en.html)
+- [Electronic Fontiers Foundation - legal cases](https://www.eff.org/cases)
 
-Copyleft licenses do not prohibit the commercial use of the copyrighted content. 
-It only prohibit you to place additional restrictions on the derivative work.
+(rr-licensing-edge)=
+### Pertinant edge cases
 
-Permissive licenses are those licenses that do not impose the reciprocity requirements.
-These licenses allow for the proprietary licensing of derivative works.
+(rr-licensing-edge-clas)=
+#### Contributor Licence Agreements
 
-A more detailed explanation is given in the chapter on license for software. {ref}`rr-licensing-software-permissive`.
+The holder of the copyright on a copy-left project can still re-licence that project or dual-licence that project under a different licence, for example to grant exclusive rights to commercially distribute that project with proprietary extensions or to make future versions proprietary.
+In a large community developed project, this would require the consent of all contributors, as they each own the copyright to their contributions.
+To get around this, some copy-left projects developed by companies that commercially licence proprietary extensions to these projects ask their contributors to sign contributor licence agreements (CLAs) which may assign the contributor's copyright to the company, or include other provisions so that they can legally dual-licence the project.
```

Comment:
I think it is also becoming quite common for large projects to have a CLA "just in case" they want to change license, copy-left or not.

I'm fairly sure Canonical and Microsoft do.

Reply:
Yes. I'm not a huge fan of this as it can, in my view, constitute a form of openwashing. You can put the (A)GPL icon on your github whilst undermining one of the mechanisms which keeps the codebase open. They often privilege one party with the rights to profit from a distribution of a version with proprietary extensions. This introduces a disincentive for upstreaming features into the core version if they can instead be offered as proprietary add-ons to increase the value proposition of their paid services. CLAs are varied and some terms would be worse than others, a CLA which places limits on the ability and reasons for re-license/dual license might be less of a issue than one which gives complete freedom to the project steward to do with user contributions as they will.

The 'open-core' model in general has a lot of inherent tension between the project and the product. The following produce less tension in my view:

- Service contracts for on-prem hosting
- Hosted instances of the open project as that guarantees complete feature parity with other deployments (save some configuration of external integrations and scale optimisations in deployment).
- The option for bespoke work on particular features as long as it get's up-streamed, including bounties

Though they admittedly do not work equally well for all projects.

Was there something specific that you though I should include here about the 'just in case' approach to CLAs?